### DISCUSSION TITLE
posts/tls-cryptography-7-diffie-hellman/

### DISCUSSION THREAD

**Discussion by giscus** [humminglab/blog.humminglab.io#8] [Announcements]
# posts/tls-cryptography-7-diffie-hellman/

지난 번에 설명한 이산대수를 이용하여 Diffie-Hellman Key Exchange(DHKE, 키교환 또는 키합의)을 이해해 보자.
TLS 암호화 채널을 절차를 간단하게 보면 다음과 같다.
서버를 믿을 수 있는지 검증, 필요시 클라이언트도 인증 RSA 암호화 채널로 키를 전달하거나, Diffie-Hellman 방식으로 키교환 교환한 키로 AES, ChaCha20와 같은 대칭키로 암호화 Diffie-Hellman은 여기서 두번째 대칭키를 교환하는 방법이다.
RSA와 같은 비대칭키를 이용하여 암호화 채널을 만든 후 이를 이용하여 대칭키를 전달하는 것은 직관적이다. 하지만 Diffie-Hellman 방식은 이와 같은 암호화 채널 없이도, 서로 키를 교환할 수 있는 방식이다.

https://blog.humminglab.io/posts/tls-cryptography-7-diffie-hellman/

**Comment by dalinaum** [comment]
오타가 있습니다.

$1895^{1721} \mod 2017$: 1427

아래로 바꾸어야 합니다.

$1859^{1721} \mod 2017$: 1427

**Reply by aqwerf** [reply]
다시 보니 기호도 잘못된 곳이 몇군데 있어서, 같이 수정하였습니다. 
잘못된 곳을 찾아 주셔서 감사합니다. ^^

**Reply by dalinaum** [reply]
양질의 글을 이렇게 읽을 수 있어 좋았습니다. 저도 감사합니다.

**Comment by ChangWanHong** [comment]
sequenceDiagram 부분 mermaid 포맷 같은데 렌더링이 안되어있습니다.

업무로 관련 내용을 찾다가 이 글을 읽게되었습니다. 저술해주신 시리즈 덕분에 업무 뿐 만 아니라 저에게도 큰 도움이 되었습니다. 감사합니다.

**Reply by aqwerf** [reply]
도움이 되었다니 저도 기분이 좋아지네요 ^^
잘못된 부분 알려주셔서 고맙습니다. 버전이 달라서인지 "^" 가 있으면 렌더링이 안되었네요. 수정하였습니다.

**Reply by ChangWanHong** [reply]
추가로 8,9,14장에서도 같은 문제가 있는것같습니다. 대응 감사합니다 ㅎ
